app/src/org/commcare/android/util/MarkupUtil.java

```diff
@@ -1,19 +1,177 @@
 package org.commcare.android.util;
 
+import in.uncod.android.bypass.Bypass;
 import net.nightwhistler.htmlspanner.HtmlSpanner;
+
+import org.commcare.dalvik.application.CommCareApp;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.dalvik.preferences.DeveloperPreferences;
+import org.javarosa.core.services.locale.Localization;
+
+import android.content.Context;
+import android.text.Html;
 import android.text.Spannable;
+import android.text.SpannableString;
 
 public class MarkupUtil {
     static HtmlSpanner htmlspanner = new HtmlSpanner();
-    
+
+    /*
+     * Developer Preference helper classes
+     */
+
+    public static Spannable styleSpannable(Context c, String message){
+
+        if(DeveloperPreferences.isCssEnabled()){
+            
+            return htmlspanner.fromHtml(MarkupUtil.getStyleString() + message);
+        }
+
+        return new SpannableString(generateMarkdown(c, message));
+
+    }
+
+    public static Spannable localizeStyleSpannable(Context c, String localizationKey){
+
+        if(DeveloperPreferences.isCssEnabled()){
+            
+            return htmlspanner.fromHtml(MarkupUtil.getStyleString() + new String(Localization.get(localizationKey)));
+        }
+
+        return new SpannableString(localizeGenerateMarkdown(c, localizationKey));
+    }
+
+    public static Spannable localizeStyleSpannable(Context c, String localizationKey, String localizationArg){
+
+        if(DeveloperPreferences.isCssEnabled()){
+            return htmlspanner.fromHtml(MarkupUtil.getStyleString() + Localization.get(localizationKey, localizationArg));
+        }
+
+        return new SpannableString(localizeGenerateMarkdown(c, localizationKey, new String[] {localizationArg}));
+    }
+
+
+
+    public static Spannable localizeStyleSpannable(Context c, String localizationKey, String[] localizationArgs){
+        if(DeveloperPreferences.isCssEnabled()){
+            return htmlspanner.fromHtml(MarkupUtil.getStyleString() + Localization.get(localizationKey, localizationArgs));
+        }
+
+        return new SpannableString(localizeGenerateMarkdown(c, localizationKey, localizationArgs));
+    }
+
+    /*
+     * Markdown styling utils from Bypass
+     * https://github.com/Uncodin/bypass
+     */
+
+    public static CharSequence localizeGenerateMarkdown(Context c, String localizationKey){
+        CharSequence mSequence = generateMarkdown(c, new String(Localization.get(localizationKey)));
```

Comment:
Will: Did you make/commit the changes to separate out markdown as a different text form and only apply the bypass stuff if we have a "markdown" form of text for a field?